### PR TITLE
Casey's accumulated nitpicks

### PR DESCRIPTION
**Pull request by CaseyCarter**
Some minor things that aren't worth individual PRs or CIs. These are almost entirely non-functional changes.

* Pretty clang-format find/version check messages: Let's make these more consistent with the messages for `find_package(Python)` and the assembler. Showing the path may help folks with problems.
* Remove extraneous `element_type` from `ContiguousIterator`: While we're here, let's use `value_type` and `difference_type` where appropriate so we can see which `int` is which.
* Remove unused include from `P2163R3_invoke_r`: This test doesn't need `is_permissive.hpp` after #4914.
* Investigate a couple of libc++ failures.
* Add some missing `#endif` comments: These were incorrectly removed by #4944.
* Change "ASAN" to the canonical "ASan" in `expected_results.txt` comments.
* Remove an imprecise comment on the primary template of `formatter` suggesting that a deleted default constructor is sufficient for a disabled specialization.
* Cleanup unintended test configuration names. The internal "Runall" test runner interprets trailing comments on configuration lines as providing a name to refer to that configuration in run reports. The STL conventionally avoids such names, forcing Runall to tell us the actual command line instead.

### CODE REVIEW

File: stl/inc/xutility [hunk 1]
```diff
@@ -435,8 +435,8 @@ struct _Get_rebind_alias<_Ty, _Other, void_t<typename _Ty::template rebind<_Othe
 
 #if _HAS_CXX20
 _EXPORT_STD template <class _Ty, class... _Types>
-    requires requires(_Ty* _Location, _Types&&... _Args) {
-        ::new (static_cast<void*>(_Location)) _Ty(_STD forward<_Types>(_Args)...); // per LWG-3888
+    requires requires(void* _VoidPtr, _Types&&... _Args) {
+        ::new (_VoidPtr) _Ty(_STD forward<_Types>(_Args)...); // per LWG-3888
```
Comment (frederick-vs-ja) [inline stl/inc/xutility:439]: The lengthy `static_cast<void*>(_Location)` was intentional. It was originally added in #3475 to speculatively implement LWG-3888. The simplified constraints conform to current standard specification, but have the speculative resolution of LWG-3888 dropped.

**Comment by AlexGuteniev** [comment]
Another candidate https://github.com/microsoft/STL/pull/4939#discussion_r1747581442

File: stl/inc/xutility (hunk 1 above)
Comment (StephanTLavavej) [reply stl/inc/xutility:439]: It appears that the `P0784R7_library_support_for_more_constexpr_containers` failure is related to this.

Comment (CaseyCarter) [reply stl/inc/xutility]: Thanks for adding the test to keep me from breaking this.

**Comment by CaseyCarter** [comment]
> Another candidate [#4939 (comment)](https://github.com/microsoft/STL/pull/4939#discussion_r1747581442) (+ `_floats` test)

Thanks for the reminder! I had a commit for this on another fork that I hadn't pushed.

File: stl/inc/xutility (hunk 1 above)
Comment (AlexGuteniev) [reply stl/inc/xutility]: It is still in the description

Comment (CaseyCarter) [reply stl/inc/xutility]: > It is still in the description

Fixed.

**Review by StephanTLavavej — APPROVED** [review]

**Comment by StephanTLavavej** [comment]
I'm mirroring this to the MSVC-internal repo (if I can chain it to the toolset update without causing problems) - please notify me if any further changes are pushed.

**Comment by StephanTLavavej** [comment]
Thanks for keeping the codebase neat and tidy! :gem: :sparkles: :heart_eyes_cat:

**Pull request merged**